### PR TITLE
Support ClusterIP and NodePort service configurations for RabbitMQ

### PR DESCRIPTION
**Pull request by henders**
Adds support for specifying the Kubernetes Service Type. Previously only the default `ClusterIP` type was supported which doesn't work with MiniKube.

Output when starting this chart with NodePort type set is now:
```bash
NOTES:

** P.S. RabbitMQ may take a few minutes to become available. Please be patient. **

The RabbitMQ AMQP port 5672 can be accessed on the following DNS name from within your cluster: ugly-armadillo-rabbitmq.default.svc.cluster.local

  Username      : user
  Password      : $(kubectl get secret --namespace default ugly-armadillo-rabbitmq -o jsonpath="{.data.rabbitmq-password}" | base64 --decode)
  ErLang Cookie : $(kubectl get secret --namespace default ugly-armadillo-rabbitmq -o jsonpath="{.data.rabbitmq-erlang-cookie}" | base64 --decode)

To access RabbitMQ from outside the cluster:
  export RABBITMQ_HOST=$(kubectl get nodes --namespace default -o jsonpath='{.items[0].status.addresses[0].address}')

  # For the AMQP port:
  export RABBITMQ_PORT=$(kubectl get svc --namespace default ugly-armadillo-rabbitmq -o jsonpath='{.spec.ports[?(@.name=="ampq")].nodePort}')
  export RABBITMQ_URL=http://$RABBITMQ_HOST:$RABBITMQ_PORT

  # For the Management Interface:
  export RABBITMQ_PORT=$(kubectl get svc --namespace default ugly-armadillo-rabbitmq -o jsonpath='{.spec.ports[?(@.name=="stats")].nodePort}')
  echo "navigate to: http://$RABBITMQ_HOST:$RABBITMQ_PORT"
```

And with ClusterIP set:
```bash
NOTES:

** P.S. RabbitMQ may take a few minutes to become available. Please be patient. **

The RabbitMQ AMQP port 5672 can be accessed on the following DNS name from within your cluster: trendy-racoon-rabbitmq.default.svc.cluster.local

  Username      : user
  Password      : $(kubectl get secret --namespace default trendy-racoon-rabbitmq -o jsonpath="{.data.rabbitmq-password}" | base64 --decode)
  ErLang Cookie : $(kubectl get secret --namespace default trendy-racoon-rabbitmq -o jsonpath="{.data.rabbitmq-erlang-cookie}" | base64 --decode)

To access RabbitMQ from outside the cluster:
  export POD_NAME=$(kubectl get pods --namespace default -l "app=trendy-racoon-rabbitmq" -o jsonpath="{.items[0].metadata.name}")

  # For the AMQP port:
  export RABBITMQ_URL=http://127.0.0.1:5672
  kubectl port-forward $POD_NAME 5672:5672

  # For the Management Interface on http://127.0.0.1:15672
  kubectl port-forward $POD_NAME 15672:15672
```

### CODE REVIEW

**Comment by henders** [comment]
Hi, I've signed the CLA!

**Comment by unguiculus** [comment]
@k8s-bot ok to test

**Comment by henders** [comment]
pinging @unguiculus @sameersbn for review, thanks.

**Comment by lachie83** [comment]
Can you bump the chart version please

**Comment by henders** [comment]
Thanks @unguiculus for bumping! 🙇

**Comment by unguiculus** [comment]
Trying to test it on GKE. The pod does not get into ready state. Any ideas?

```
$ stern rabbitmq
+ hazy-ladybug-rabbitmq-3189215870-6whr5 › hazy-ladybug-rabbitmq
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq Welcome to the Bitnami rabbitmq container
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-rabbitmq
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-rabbitmq/issues
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq Send us your feedback at containers@bitnami.com
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq nami    INFO  Initializing rabbitmq
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq rabbitm INFO  Creating RabbitMQ System User...
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq rabbitm INFO  Setting RabbitMQ environment...
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq rabbitm INFO  Starting RabbitMQ node rabbit@localhost...
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq rabbitm INFO  Configuring RabbitMQ permissions...
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq rabbitm INFO
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq rabbitm INFO  ########################################################################
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq rabbitm INFO   Installation parameters for rabbitmq:
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq rabbitm INFO     Persisted data and properties have been restored.
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq rabbitm INFO     Any input specified will not take effect.
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq rabbitm INFO   (Passwords are not shown for security reasons)
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq rabbitm INFO  ########################################################################
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq rabbitm INFO
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq nami    INFO  rabbitmq successfully initialized
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq INFO  ==> Starting rabbitmq...
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq              {table_missing,rabbit_runtime_parameters},
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq              {table_missing,rabbit_durable_queue},
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq              {table_missing,rabbit_queue},
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq              {table_missing,gm_group},
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq              {table_missing,mirrored_sup_childspec}]},
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq         {rabbit,start,[normal,[]]}}}
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq Log files (may contain more information):
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq    /opt/bitnami/rabbitmq/var/log/rabbitmq/rabbit@localhost.log
hazy-ladybug-rabbitmq-3189215870-6whr5 hazy-ladybug-rabbitmq    /opt/bitnami/rabbitmq/var/log/rabbitmq/rabbit@localhost-sasl.log
```

```
kubectl describe pod hazy-ladybug-rabbitmq
Name:		hazy-ladybug-rabbitmq-3189215870-6whr5
Namespace:	default
Node:		gke-plasma-k8s-default-pool-c68bc53c-1p48/10.133.0.4
Start Time:	Tue, 13 Jun 2017 21:32:08 +0200
Labels:		app=hazy-ladybug-rabbitmq
		pod-template-hash=3189215870
Annotations:	kubernetes.io/created-by={"kind":"SerializedReference","apiVersion":"v1","reference":{"kind":"ReplicaSet","namespace":"default","name":"hazy-ladybug-rabbitmq-3189215870","uid":"43141c44-506e-11e7-b54f...
Status:		Running
IP:		10.28.2.22
Controllers:	ReplicaSet/hazy-ladybug-rabbitmq-3189215870
Containers:
  hazy-ladybug-rabbitmq:
    Container ID:	docker://5c06a0142e14135987e5668b69d5604ae94a7bb0eb5872038f940630dbc5bd19
    Image:		bitnami/rabbitmq:3.6.10-r0
    Image ID:		docker://sha256:02603d48b447189e7a7fb82a2453b8cecaa43b214b25873f504cb4a4609074be
    Ports:		4369/TCP, 5672/TCP, 25672/TCP, 15672/TCP
    State:		Running
      Started:		Tue, 13 Jun 2017 21:39:31 +0200
    Last State:		Terminated
      Reason:		Completed
      Exit Code:	0
      Started:		Mon, 01 Jan 0001 00:00:00 +0000
      Finished:		Tue, 13 Jun 2017 21:39:31 +0200
    Ready:		False
    Restart Count:	3
    Requests:
      cpu:	100m
      memory:	256Mi
    Liveness:	exec [rabbitmqctl status] delay=120s timeout=5s period=10s #success=1 #failure=6
    Readiness:	exec [rabbitmqctl status] delay=10s timeout=3s period=5s #success=1 #failure=3
    Environment:
      RABBITMQ_USERNAME:		user
      RABBITMQ_PASSWORD:		<set to the key 'rabbitmq-password' in secret 'hazy-ladybug-rabbitmq'>		Optional: false
      RABBITMQ_ERLANG_COOKIE:		<set to the key 'rabbitmq-erlang-cookie' in secret 'hazy-ladybug-rabbitmq'>	Optional: false
      RABBITMQ_NODE_PORT:		5672
      RABBITMQ_NODE_TYPE:		stats
      RABBITMQ_NODE_NAME:		rabbit@localhost
      RABBITMQ_CLUSTER_NODE_NAME:
      RABBITMQ_VHOST:			/
      RABBITMQ_MANAGER_PORT:		15672
    Mounts:
      /bitnami/rabbitmq from data (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-l8gkg (ro)
Conditions:
  Type		Status
  Initialized 	True
  Ready 	False
  PodScheduled 	True
Volumes:
  data:
    Type:	PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:	hazy-ladybug-rabbitmq
    ReadOnly:	false
  default-token-l8gkg:
    Type:	Secret (a volume populated by a Secret)
    SecretName:	default-token-l8gkg
    Optional:	false
QoS Class:	Burstable
Node-Selectors:	<none>
Tolerations:	node.alpha.kubernetes.io/notReady=:Exists:NoExecute for 300s
		node.alpha.kubernetes.io/unreachable=:Exists:NoExecute for 300s
Events:
  FirstSeen	LastSeen	Count	From							SubObjectPath				Type		Reason		Message
  ---------	--------	-----	----							-------------				--------	------		-------
  9m		9m		1	default-scheduler										Normal		Scheduled	Successfully assigned hazy-ladybug-rabbitmq-3189215870-6whr5 to gke-plasma-k8s-default-pool-c68bc53c-1p48
  9m		9m		1	kubelet, gke-plasma-k8s-default-pool-c68bc53c-1p48	spec.containers{hazy-ladybug-rabbitmq}	Normal		Created		Created container with id 7c1048b48c37af14c400862c4b5c083b4f6399ec56a07a21ec209727ca73d414
  9m		9m		1	kubelet, gke-plasma-k8s-default-pool-c68bc53c-1p48	spec.containers{hazy-ladybug-rabbitmq}	Normal		Started		Started container with id 7c1048b48c37af14c400862c4b5c083b4f6399ec56a07a21ec209727ca73d414
  6m		6m		1	kubelet, gke-plasma-k8s-default-pool-c68bc53c-1p48	spec.containers{hazy-ladybug-rabbitmq}	Normal		Created		Created container with id 519e0544728109483c548998a918834c698d1c4fe8349765f5ef66e9edaa1811
  6m		6m		1	kubelet, gke-plasma-k8s-default-pool-c68bc53c-1p48	spec.containers{hazy-ladybug-rabbitmq}	Normal		Killing		Killing container with id docker://7c1048b48c37af14c400862c4b5c083b4f6399ec56a07a21ec209727ca73d414:pod "hazy-ladybug-rabbitmq-3189215870-6whr5_default(fd538d75-506e-11e7-b54f-42010a840267)" container "hazy-ladybug-rabbitmq" is unhealthy, it will be killed and re-created.
  6m		6m		1	kubelet, gke-plasma-k8s-default-pool-c68bc53c-1p48	spec.containers{hazy-ladybug-rabbitmq}	Normal		Started		Started container with id 519e0544728109483c548998a918834c698d1c4fe8349765f5ef66e9edaa1811
  3m		3m		1	kubelet, gke-plasma-k8s-default-pool-c68bc53c-1p48	spec.containers{hazy-ladybug-rabbitmq}	Normal		Killing		Killing container with id docker://519e0544728109483c548998a918834c698d1c4fe8349765f5ef66e9edaa1811:pod "hazy-ladybug-rabbitmq-3189215870-6whr5_default(fd538d75-506e-11e7-b54f-42010a840267)" container "hazy-ladybug-rabbitmq" is unhealthy, it will be killed and re-created.
  3m		3m		1	kubelet, gke-plasma-k8s-default-pool-c68bc53c-1p48	spec.containers{hazy-ladybug-rabbitmq}	Normal		Created		Created container with id 38959d882dca6c24ff38b5f2452002691b10d7bba29ff755b05739dd2d3c4e6d
  3m		3m		1	kubelet, gke-plasma-k8s-default-pool-c68bc53c-1p48	spec.containers{hazy-ladybug-rabbitmq}	Normal		Started		Started container with id 38959d882dca6c24ff38b5f2452002691b10d7bba29ff755b05739dd2d3c4e6d
  9m		1m		4	kubelet, gke-plasma-k8s-default-pool-c68bc53c-1p48	spec.containers{hazy-ladybug-rabbitmq}	Normal		Pulled		Container image "bitnami/rabbitmq:3.6.10-r0" already present on machine
  1m		1m		1	kubelet, gke-plasma-k8s-default-pool-c68bc53c-1p48	spec.containers{hazy-ladybug-rabbitmq}	Normal		Killing		Killing container with id docker://38959d882dca6c24ff38b5f2452002691b10d7bba29ff755b05739dd2d3c4e6d:pod "hazy-ladybug-rabbitmq-3189215870-6whr5_default(fd538d75-506e-11e7-b54f-42010a840267)" container "hazy-ladybug-rabbitmq" is unhealthy, it will be killed and re-created.
  1m		1m		1	kubelet, gke-plasma-k8s-default-pool-c68bc53c-1p48	spec.containers{hazy-ladybug-rabbitmq}	Normal		Created		Created container with id 5c06a0142e14135987e5668b69d5604ae94a7bb0eb5872038f940630dbc5bd19
  1m		1m		1	kubelet, gke-plasma-k8s-default-pool-c68bc53c-1p48	spec.containers{hazy-ladybug-rabbitmq}	Normal		Started		Started container with id 5c06a0142e14135987e5668b69d5604ae94a7bb0eb5872038f940630dbc5bd19
```

File: stable/rabbitmq/templates/NOTES.txt [hunk 1]
```diff
@@ -3,12 +3,31 @@
 
 The RabbitMQ AMQP port {{ default "5672" .Values.rabbitmqNodePort }} can be accessed on the following DNS name from within your cluster: {{ template "fullname" . }}.{{ .Release.Namespace }}.svc.cluster.local
 
-  echo Username      : {{ .Values.rabbitmqUsername }}
-  echo Password      : $(kubectl get secret --namespace {{ .Release.Namespace }} {{ template "fullname" . }} -o jsonpath="{.data.rabbitmq-password}" | base64 --decode)
-  echo ErLang Cookie : $(kubectl get secret --namespace {{ .Release.Namespace }} {{ template "fullname" . }} -o jsonpath="{.data.rabbitmq-erlang-cookie}" | base64 --decode)
+  Username      : {{ .Values.rabbitmqUsername }}
+  Password      : $(kubectl get secret --namespace {{ .Release.Namespace }} {{ template "fullname" . }} -o jsonpath="{.data.rabbitmq-password}" | base64 --decode)
+  ErLang Cookie : $(kubectl get secret --namespace {{ .Release.Namespace }} {{ template "fullname" . }} -o jsonpath="{.data.rabbitmq-erlang-cookie}" | base64 --decode)
```
Comment (sameersbn) [inline stable/rabbitmq/templates/NOTES.txt:8]: I suggest reverting the changes here to restore the `echo` command invocations. Copy pasting the instructions from the NOTES is just so much better with the `echo` commands

File: stable/rabbitmq/templates/NOTES.txt [hunk 2]
```diff
@@ -3,12 +3,31 @@
...
+  # For the Management Interface:
+  export RABBITMQ_PORT=$(kubectl get svc --namespace {{ .Release.Namespace }} {{ template "fullname" . }} -o jsonpath='{.spec.ports[?(@.name=="stats")].nodePort}')
+  echo "navigate to: http://$RABBITMQ_HOST:$RABBITMQ_PORT"
+
+{{- else if contains "ClusterIP" .Values.service.type }}
   export POD_NAME=$(kubectl get pods --namespace {{ .Release.Namespace }} -l "app={{ template "fullname" . }}" -o jsonpath="{.items[0].metadata.name}")
-  echo URL : http://127.0.0.1:{{ default "15672" .Values.rabbitmqManagerPort }}
+
+  # For the AMQP port:
+  export RABBITMQ_URL=http://127.0.0.1:{{ default "5672" .Values.rabbitmqNodePort }}
```
Comment (sameersbn) [inline stable/rabbitmq/templates/NOTES.txt:27]: Perhaps you could remove the `export` and change this line to `# For the APMQ port on amqp://127.0.0.1:{{ default "5672" .Values.rabbitmqNodePort }}`

File: stable/rabbitmq/templates/NOTES.txt [hunk 3]
```diff
@@ -3,12 +3,31 @@
...
+  kubectl port-forward $POD_NAME {{ default "5672" .Values.rabbitmqNodePort }}:{{ default "5672" .Values.rabbitmqNodePort }}
+
+  # For the Management Interface on http://127.0.0.1:{{ default "15672" .Values.rabbitmqManagerPort }}
   kubectl port-forward $POD_NAME {{ default "15672" .Values.rabbitmqManagerPort }}:{{ default "15672" .Values.rabbitmqManagerPort }}
```
Comment (sameersbn) [inline stable/rabbitmq/templates/NOTES.txt:31]: Maybe we should combine these into a single command:
```
kubectl port-forward $POD_NAME {{ default "5672" .Values.rabbitmqNodePort }}:{{ default "5672" .Values.rabbitmqNodePort }} {{ default "15672" .Values.rabbitmqManagerPort }}:{{ default "15672" .Values.rabbitmqManagerPort }}
```

WDYT?

File: stable/rabbitmq/templates/NOTES.txt [hunk 4]
```diff
@@ -3,12 +3,31 @@
...
 
-To Access the RabbitMQ Management interface:
+To access RabbitMQ from outside the cluster:
 
+{{- if contains "NodePort" .Values.service.type }}
+  export RABBITMQ_HOST=$(kubectl get nodes --namespace {{ .Release.Namespace }} -o jsonpath='{.items[0].status.addresses[0].address}')
+
+  # For the AMQP port:
+  export RABBITMQ_PORT=$(kubectl get svc --namespace {{ .Release.Namespace }} {{ template "fullname" . }} -o jsonpath='{.spec.ports[?(@.name=="ampq")].nodePort}')
+  export RABBITMQ_URL=http://$RABBITMQ_HOST:$RABBITMQ_PORT
```
Comment (sameersbn) [inline stable/rabbitmq/templates/NOTES.txt:17]: I would suggest removing the `export`'s and change this line to # For the APMQ port on `amqp://$RABBITMQ_HOST:$RABBITMQ_PORT`

File: stable/rabbitmq/templates/NOTES.txt [hunk 5]
```diff
@@ -3,12 +3,31 @@
...
+
+  # For the Management Interface:
+  export RABBITMQ_PORT=$(kubectl get svc --namespace {{ .Release.Namespace }} {{ template "fullname" . }} -o jsonpath='{.spec.ports[?(@.name=="stats")].nodePort}')
+  echo "navigate to: http://$RABBITMQ_HOST:$RABBITMQ_PORT"
+
+{{- else if contains "ClusterIP" .Values.service.type }}
```
Comment (sameersbn) [inline stable/rabbitmq/templates/NOTES.txt:23]: Could you also please add instructions for `LoadBalancer` service type?

**Review by sameersbn — CHANGES_REQUESTED** [review]

**Comment by unguiculus** [comment]
Marking as stale. Please update within the next week.

**Comment by viglesiasce** [comment]
Closing as stale please re-open if/when you have a chance to work on this.

**Pull request closed without merging**